### PR TITLE
[PF-246] Update logger to use spring cloud json format

### PR DESCRIPTION
**Pull request by yonghaoy**
Copied from https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/src/main/resources/logback-spring.xml

After that, we are able to use Error Reporting: 
https://console.cloud.google.com/errors/CImIqKDupaT8mgE?time=PT1H&project=terra-kernel-k8s&supportedpurview=project

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 buffer:
   db:
     password:
-    recreate-db-on-start: false
+    recreate-db-on-start: true
```
Comment (wchamber) [inline src/main/resources/application.yml:4]: uh, is this intentional?

**Review by wchamber — COMMENTED** [review]
Maybe add Zach for an additional sanity check since I think he set this stuff up for WM?
LGTM other than recreate-db-on-start

**Review by zloery — APPROVED** [review]
LGTM though I haven't looked at any actual logs - I'd verify that logs for an integration deployment (if you have one) look right on Stackdriver, and that any errors show up in Error Reporting

**Comment by yonghaoy** [comment]
> LGTM though I haven't looked at any actual logs - I'd verify that logs for an integration deployment (if you have one) look right on Stackdriver, and that any errors show up in Error Reporting

Thanks! 
I append a sample in the PR's description. 
Also, here is a [sample log](https://console.cloud.google.com/logs/viewer?interval=P1D&project=terra-kernel-k8s&supportedpurview=project&minLogLevel=0&expandAll=false&timestamp=2020-12-03T17%3A09%3A59.050000000Z&customFacets&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.cluster_name%3D%22terra-integration%22%0Aresource.labels.container_name%3D%22terra-buffer%22%0Aresource.labels.location%3D%22us-central1-a%22%0Aresource.labels.namespace_name%3D%22terra-yyu%22%0Aresource.labels.project_id%3D%22terra-kernel-k8s%22%0Aresource.labels.pod_name%3D%22terra-buffer-deployment-54d97cf74b-dqvgz%22%0Atimestamp%3D%222020-12-03T00%3A10%3A36.466000000Z%22%0AinsertId%3D%22qeyx4nd461yie4u7v%22&scrollTimestamp=2020-12-03T00%3A10%3A36.466000000Z&dateRangeStart=2020-12-02T17%3A09%3A59.883Z&dateRangeEnd=2020-12-03T17%3A09%3A59.883Z)

**Review by wchamber — APPROVED** [review]

**Pull request merged**